src/hotspot/share/opto/output.cpp

```diff
@@ -1158,6 +1172,12 @@ void PhaseOutput::Process_OopMap_Node(MachNode *mach, int current_offset) {
 
         for (int j = 0; j< merge->possible_objects()->length(); j++) {
           ObjectValue* ov = merge->possible_objects()->at(j)->as_ObjectValue();
+          // Already flagged as 'root' by something else. We shouldn't change it
+          // to non-root in a younger JVMS because it may need to be alive in
+          // the younger JVMS.
+          if (ov->is_root()) {
+            continue;
```

Comment:
You can either fuse `ov->is_root()` check into `is_root` computation (`bool is_root = ov->is_root() || ...`) or turn it into an `if-then-else` (`if (ov->is_root()) { /* comment */ } else { bool is_root = ...;  ov->set_root(is_root); }`). I find both cases easier to read.

Reply:
Done. Thanks Vladimir.